File: FrontEnd/food_order/lib/view_models/Promotions/PromotionVM.dart
import 'package:food_delivery/view_models/Foods/FoodVM.dart';
import 'package:json_annotation/json_annotation.dart';

part 'PromotionVM.g.dart';

@JsonSerializable()
class PromotionVM {
  PromotionVM();
  late int id;
  late String name;
  late String code;
  String? desciption;
  late DateTime startDate;
  late DateTime endDate;
  int? useTimes;
  late double percent;
  late bool enabled;
  double? max;
  double? minPrice;
  late bool isGlobal;
  late int timeUsedByCurrentUser;
  late String imagePath;

  @JsonKey(ignore: true)
  late List<FoodVM> foodVMs;

  factory PromotionVM.fromJson(Map<String, dynamic> json) =>
      _$PromotionVMFromJson(json);
  Map<String, dynamic> toJson() => _$PromotionVMToJson(this);
}

File: BackEnd/FoodOrder.Core/ViewModels/Promotions/PromotionVM.cs
using FoodOrder.Core.ViewModels.Foods;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoodOrder.Core.ViewModels.Promotions
{
    public class PromotionVM
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        public string Desciption { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int UseTimes { get; set; }
        public float Percent { get; set; }
        public bool Enabled { get; set; }
        public double Max { get; set; }
        public double MinPrice { get; set; }
        public bool IsGlobal { get; set; }
        public int Priority { get; set; }
        public int TimeUsedByCurrentUser { get; set; }
    }
}
